waspc/src/Wasp/Project/Analyze.hs

```diff
@@ -118,15 +232,15 @@ constructAppSpec waspDir options packageJson parsedPrismaSchema decls = do
 
   return $ runValidation ASV.validateAppSpec appSpec
 
-findWaspFile :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs File'))
+findWaspFile :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe WaspFile)
 findWaspFile waspDir = do
   files <- fst <$> IOUtil.listDirectory waspDir
-  return $ (waspDir </>) <$> find isWaspFile files
+  return $ findWaspTsFile files <|> findWaspLangFile files
```

Comment:
> Should I flip this around? In other words, only declare main.wasp.ts as the valid config file if main.wasp is missing?

IMHO I don't think it's that important and what we have now is fine for this stage.

Reply:
I think it is good enough. Question of order is interesting, although best would probably be to throw error if there are both files (.wasp and .wasp.mts) present, because we don't want to guess. So we could say, you can't have both, there needs to be a single definition, pick your poison!

Reply:
This is a great idea, I'll do that (too bad I won't be able to use `<|>` though).

Reply:
@sodic I know you are joking about `<|>` but I thought the same thing :D.